File: app/src/main/java/com/cabe/app/novel/model/NovelList.java
package com.cabe.app.novel.model;

import android.text.TextUtils;

import java.util.List;

/**
 * 作者：沈建芳 on 2017/10/11 10:42
 */
public class NovelList extends BaseObject {
    public String title;
    public String author;
    public String update;
    public String picUrl;
    public String type;
    public String state;
    public SourceType source = SourceType.X23US;
    public List<NovelContent> list;

    public String getTips() {
        String tips = "作者：" + author;
        if(!TextUtils.isEmpty(update)) {
            tips += "  更新时间：" + update;
        }
        return tips;
    }

    public String toString() {
        return title + "(" + author + ")#" + update;
    }
}

File: app/src/main/java/com/cabe/app/novel/model/NovelList.kt
package com.cabe.app.novel.model

import android.text.TextUtils

/**
 * 作者：沈建芳 on 2017/10/11 10:42
 */
class NovelList : BaseObject() {
    var title: String? = null
    var author: String? = null
    var update: String? = null
    var lastChapter: String? = null
    var picUrl: String? = null
    var type: String? = null
    var state: String? = null
    var source = SourceType.X23US
    var list: List<NovelContent>? = null
    val tips: String
        get() {
            var tips = "作者：$author"
            if (!TextUtils.isEmpty(update)) {
                tips += "  更新时间：$update"
            }
            return tips
        }

    override fun toString(): String {
        return "$title($author)#$update"
    }
}